### PR TITLE
RS/CH/get_zcc

### PR DESCRIPTION
**Pull request by charliepnnl**

### CODE REVIEW

File: docs/ruleset_functions/get_zone_conditioning_category.md [hunk 1]
```diff
@@ -64,7 +66,7 @@ Logic:
       - For each surface in zone: `for surface in zone.surfaces:`  
 
         - Check if surface is interior, get adjacent zone: `if surface.adjacent_to == "INTERIOR": adjacent_zone = match_data_element(RMR, zones, surface.adjacent_zone_id)`  
-
+**[CH: There is no `surface.fenestration_subsurfaces` field. There is a `surface.subsurfaces` field instead. Subsurface has `opaque_area` and `glazed_area`.]**
```
Comment (xingcx) [inline docs/ruleset_functions/get_zone_conditioning_category.md:69]: Surface.fenestration_subsurfaces is no longer in the schema. I'll correct the calculation below.

File: docs/ruleset_functions/get_zone_conditioning_category.md [hunk 2]
```diff
@@ -88,7 +90,7 @@ Logic:
       - For each space in zone: `for space in zone.spaces:`  
 
         - Check if lighting space type is residential, space is classified as residential: `if space.lighting_space_type in ["DORMITORY_LIVING_QUARTERS" , "FIRE_STATION_SLEEPING_QUARTERS", "GUEST_ROOM", "DWELLING_UNIT", "HEALTHCARE_FACILITY_NURSERY", "HEALTHCARE_FACILITY_PATIENT_ROOM"]: space_residential_flag = TRUE`  
-
+**[CH: I suggest replacing `space_residential_flag` with `zone_has_residential_spaces` and replacing `space_nonresidential_flag` with `zone_has_residential_spaces`.]**
```
Comment (xingcx) [inline docs/ruleset_functions/get_zone_conditioning_category.md:93]: I'll update these.

File: docs/ruleset_functions/get_zone_conditioning_category.md [hunk 3]
```diff
@@ -107,10 +109,10 @@ Logic:
 
     - Else if zone has interior parking spaces, classify zone as unenclosed: `if（ space.lighting_space_type == "PARKING_AREA_INTERIOR" for space in zone.spaces ）: zone_conditioning_category_dict[zone.id] = "UNENCLOSED"`  
 
-    - Else if zone is crawlspace, classify zone as unenclosed: `else if ( ( zone.volume / sum( space.floor_area for space in zone.spaces ) ) < CRAWLSPACE_HEIGHT_THRESHOLD ) AND ( ( get_opaque_surface_type(surface) in ["HEATED SLAB-ON-GRADE", "UNHEATED SLAB-ON-GRADE"] ) AND ( surface.adjacent_to == "GROUND" ) for surface in zone.surfaces ): zone_conditioning_category_dict[zone.id] = "UNENCLOSED"`  (Note XC, need to check surface != Floor, RDS online not updated)
+    - Else if zone is crawlspace, classify zone as unenclosed: `else if ( ( zone.volume / sum( space.floor_area for space in zone.spaces ) ) < CRAWLSPACE_HEIGHT_THRESHOLD ) AND ( ( get_opaque_surface_type(surface) in ["HEATED SLAB-ON-GRADE", "UNHEATED SLAB-ON-GRADE"] ) AND ( surface.adjacent_to == "GROUND" ) for `**[CH: insert `any`]**` surface in zone.surfaces ): zone_conditioning_category_dict[zone.id] = "UNENCLOSED"`  (Note XC, need to check surface != Floor, RDS online not updated)
```
Comment (xingcx) [inline docs/ruleset_functions/get_zone_conditioning_category.md:112]: I'll correct this.

File: docs/ruleset_functions/get_zone_conditioning_category.md [hunk 4]
```diff
@@ -107,10 +109,10 @@ Logic:
...
 
-    - Else if zone is attic, classify zone as unenclosed: `else if ( ( get_opaque_surface_type(surface) == "CEILING" ) AND ( surface.adjacent_to == "EXTERIOR" ) for surface in zone.surfaces ): zone_conditioning_category_dict[zone.id] = "UNENCLOSED"`  
+    - Else if zone is attic, classify zone as unenclosed: `else if ( ( get_opaque_surface_type(surface) == "CEILING"`**[CH: ROOF?]**` ) AND ( surface.adjacent_to == "EXTERIOR" ) for `**[CH: insert `any`]**` surface in zone.surfaces ): zone_conditioning_category_dict[zone.id] = "UNENCLOSED"`
```
Comment (xingcx) [inline docs/ruleset_functions/get_zone_conditioning_category.md:114]: I'll correct "Ceiling" to "Roof“ and add any for surfaces.

File: docs/ruleset_functions/get_zone_conditioning_category.md [hunk 5]
```diff
@@ -19,21 +19,23 @@ Constants:
...
+  - Calculate and save total central heating output per floor area for HVAC system to dictionary: `hvac_heat_capacity_dict[hvac_sys_id] = total_central_heat_capacity / hvac_zone_list_w_area_dict[hvac_sys_id]["TOTAL_AREA"]`
 
-    - Calculate and save total central sensible cooling output per floor area for HVAC system to dictionary: `hvac_cool_capacity_dict[hvac_sys_id] = total_central_sensible_cool_capacity / hvac_zone_list_w_area_dict[hvac_sys_id]["TOTAL_AREA"]`
+- Get building climate zone: `climate_zone = RMR.weather.climate_zone`  
 
-    - Calculate and save total central heating output per floor area for HVAC system to dictionary: `hvac_heat_capacity_dict[hvac_sys_id] = total_central_heat_capacity / hvac_zone_list_w_area_dict[hvac_sys_id]["TOTAL_AREA"]`
+- Get heated space criteria: `system_min_heating_output = data_lookup(table_3_2,climate_zone)`
 
 - For each zone in RMR: `for zone in RMR...zones:`
```
Comment (xingcx) [inline docs/ruleset_functions/get_zone_conditioning_category.md:41]: I will need to include zonal cooling output in the calculation below. The schema has been updated to reflect that.

Comment (inkmug) [reply docs/ruleset_functions/get_zone_conditioning_category.md:41]: This is updated.

Comment (xingcx) [reply docs/ruleset_functions/get_zone_conditioning_category.md:41]: This is updated.

File: docs/ruleset_functions/get_zone_conditioning_category.md (hunk 1 above)
Comment (xingcx) [reply docs/ruleset_functions/get_zone_conditioning_category.md:69]: This is corrected.

File: docs/ruleset_functions/get_zone_conditioning_category.md (hunk 2 above)
Comment (xingcx) [reply docs/ruleset_functions/get_zone_conditioning_category.md:93]: This is updated.

File: docs/ruleset_functions/get_zone_conditioning_category.md (hunk 3 above)
Comment (xingcx) [reply docs/ruleset_functions/get_zone_conditioning_category.md:112]: This is corrected. Also deleted the surface.adjacent_to requirement. Heated and unheated slab-on-grades are adjacent to ground by definition.

File: docs/ruleset_functions/get_zone_conditioning_category.md (hunk 4 above)
Comment (xingcx) [reply docs/ruleset_functions/get_zone_conditioning_category.md:114]: This is corrected.

File: docs/ruleset_functions/get_zone_conditioning_category.md [hunk 6]
```diff
@@ -64,7 +66,7 @@ Logic:
...
-
+**[CH: There is no `surface.fenestration_subsurfaces` field. There is a `surface.subsurfaces` field instead. Subsurface has `opaque_area` and `glazed_area`.]**
           - If adjacent zone is directly conditioned (heated or cooled), add the product of the U-factor and surface area to the directly conditioned type: `if adjacent_zone in directly_conditioned_zone: directly_conditioned_product_sum += sum( ( fenestration.glazed_area + fenestration.opaque_area ) * fenestration.u_factor for fenestration in surface.fenestration_subsurfaces ) + ( surface.area - sum( ( fenestration.glazed_area + fenestration.opaque_area ) for fenestration in surface.fenestration_subsurfaces ) * surface.construction.u_factor`  
 
           - Else, add the product of the U-factor and surface area to the other type: `else: other_product_sum += sum( ( fenestration.glazed_area + fenestration.opaque_area ) * fenestration.u_factor for fenestration in surface.fenestration_subsurfaces ) + ( surface.area - sum( ( fenestration.glazed_area + fenestration.opaque_area ) for fenestration in surface.fenestration_subsurfaces ) * surface.construction.u_factor`
```
[the remark this answers is not in the document]
Comment (weilixu) [reply docs/ruleset_functions/get_zone_conditioning_category.md]: The decision is escalating it to code interpretation. Bur for now, let's keep it as it is for merging.

**Comment by weilixu** [comment]
Will merge the ZCC an additional ticket is created to address the ground surfaces in the product sum of heat transfer rate. #362

**Pull request merged**